x-pack/plugins/security_solution/common/api/detection_engine/signals_migration/create_signals_migration/create_signals_migration.gen.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from 'zod';
+
+/*
+ * NOTICE: Do not edit this file manually.
+ * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
+ *
+ * info:
+ *   title: Initiates alerts migration API endpoint
+ *   version: 2023-10-31
+ */
+
+import { NonEmptyString } from '../../../model/primitives.gen';
+
+export type AlertsIndexMigrationSuccess = z.infer<typeof AlertsIndexMigrationSuccess>;
+export const AlertsIndexMigrationSuccess = z.object({
+  index: z.string(),
+  migration_id: z.string(),
+  migration_index: z.string(),
```

Comment:
To match the API route behavior these should exist but be `null` for the error and skipped migration cases.

Reply:
@marshallmain,

OAS doesn't provide a way to specify explicit `null` fields. There is a `nullable` modifier which marks a type as nullable. In this particular case it's possible to declare a single response type instead of `AlertsIndexMigrationSuccess`, `AlertsIndexMigrationError` and `SkippedAlertsIndexMigration` with `migration_id` and `migration_index` being string or null. Such schema won't describe the response precisely neither.

What's the approach you expect here?

Reply:
We discussed on slack and the approach is ok as is. Better to leave the properties out of the response schema than have them typed incorrectly or make the rest of the schema less precise.